doc/core/pending_api_cleanups.md

```diff
@@ -0,0 +1,15 @@
+There are times when we make changes that include a temporary shim for
```

Comment:
@ctiller, here's the file you requested that I add yesterday.

@nicolasnoble, please keep this file in mind whenever the next release happens.